okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttpBuilder.scala

```diff
@@ -36,46 +35,59 @@ import okhttp3.{
 import okio.BufferedSink
 import org.http4s.{Header, Headers, HttpVersion, Method, Request, Response, Status}
 import org.http4s.client.Client
-import org.http4s.internal.{BackendBuilder, invokeCallback}
+import org.http4s.internal.BackendBuilder
 import org.http4s.internal.CollectionCompat.CollectionConverters._
 import org.log4s.getLogger
-import scala.concurrent.ExecutionContext
 import scala.util.control.NonFatal
+import scala.util.chaining._
+import cats.effect.std.Dispatcher
 
 /** A builder for [[org.http4s.client.Client]] with an OkHttp backend.
   *
-  * @define BLOCKINGEC an execution context onto which all blocking
-  * I/O operations will be shifted.
+  * @define DISPATCHER a [[cats.effect.std.Dispatcher]] using which
+  * we will call unsafeRunSync
   *
   * @define WHYNOSHUTDOWN It is assumed that the OkHttp client is
   * passed to us as a Resource, or that the caller will shut it down, or
   * that the caller is comfortable letting OkHttp's resources expire on
   * their own.
   *
   * @param okHttpClient the underlying OkHttp client.
-  * @param blockingExecutionContext $BLOCKINGEC
+  * @param dispatcher $BLOCKINGEC
   */
 sealed abstract class OkHttpBuilder[F[_]] private (
     val okHttpClient: OkHttpClient,
-    val blocker: Blocker
-)(implicit protected val F: ConcurrentEffect[F], cs: ContextShift[F])
+    val dispatcher: Dispatcher[F]
+)(implicit protected val F: Async[F])
     extends BackendBuilder[F, Client[F]] {
   private[this] val logger = getLogger
 
+  type Result[F[_]] = Either[Throwable, Resource[F, Response[F]]]
```

Comment:
Can this alias be private to reduce API liability?

Reply:
done